
# HG changeset patch
# User Andreas Regel <andreas.regel@gmx.de>
# Date 1270407373 -7200
# Node ID e63590be3520da717e9c87f0c6c63e1c921daefa
# Parent 5f42e0ec7407299f70b91aba48ee18462943f81e
stb0899: fixed reading of IF_AGC_GAIN register

When reading IF_AGC_GAIN register a wrong value for the base address register was used (STB0899_DEMOD instead of STB0899_S2DEMOD).

Signed-off-by: Andreas Regel <andreas.regel@gmx.de>

--- a/linux/drivers/media/dvb/frontends/stb0899_drv.c	Sun Apr 04 20:51:07 2010 +0200
+++ b/linux/drivers/media/dvb/frontends/stb0899_drv.c	Sun Apr 04 20:56:13 2010 +0200
@@ -1029,7 +1029,7 @@
 		break;
 	case SYS_DVBS2:
 		if (internal->lock) {
-			reg = STB0899_READ_S2REG(STB0899_DEMOD, IF_AGC_GAIN);
+			reg = STB0899_READ_S2REG(STB0899_S2DEMOD, IF_AGC_GAIN);
 			val = STB0899_GETFIELD(IF_AGC_GAIN, reg);
 
 			*strength = stb0899_table_lookup(stb0899_dvbs2rf_tab, ARRAY_SIZE(stb0899_dvbs2rf_tab) - 1, val);

